HW2/P4/driver.py

```diff
@@ -20,8 +20,15 @@ def py_median_3x3(image, iterations=10, num_threads=1):
     tmpA = image.copy()
     tmpB = np.empty_like(tmpA)
 
-    for i in range(iterations):
-        filtering.median_3x3(tmpA, tmpB, 0, 1)
+    for i in xrange(iterations):
+        thr_vec = []
+        for threadidx in xrange(num_threads):
+            # operate with more than one thread
+            thr = threading.Thread(target=filtering.median_3x3, args=(tmpA, tmpB, threadidx, num_threads))
+            thr.start()
+            thr_vec.append(thr)
+        # join the threads
+        for thr in thr_vec: thr.join()
```

Comment:
Very nice and clean! It's an interesting approach and it seems that no locks or conditional variables are used.